ext/opentelemetry-exporter-cloud-trace/tests/test_cloud_trace_propagator.py

```diff
@@ -0,0 +1,217 @@
+        self.valid_span_id = 17725314949316355921
+        self.too_long_id = 111111111111111111111111111111111111111111111
+
+    def _extract(self, header_value):
+        """Test helper"""
+        header = {_TRACE_CONTEXT_HEADER_NAME: [header_value]}
+        new_context = self.propagator.extract(get_dict_value, header)
+        return trace.get_current_span(new_context).get_context()
+
+    def _inject(self, span=None):
+        """Test helper"""
+        ctx = get_current()
+        if span:
+            ctx = trace.set_span_in_context(span, ctx)
+        output = {}
+        self.propagator.inject(dict.__setitem__, output, context=ctx)
+        return output.get(_TRACE_CONTEXT_HEADER_NAME)
+
+    def test_no_context_header(self):
+        header = {}
+        new_context = self.propagator.extract(get_dict_value, header)
+        self.assertEqual(
+            trace.get_current_span(new_context).get_context(),
+            trace.INVALID_SPAN.get_context(),
+        )
+
+    def test_empty_context_header(self):
+        header = ""
+        self.assertEqual(
+            self._extract(header), trace.INVALID_SPAN.get_context()
+        )
+
+    def test_valid_header(self):
+        header = "{}/{};o=1".format(
+            get_hexadecimal_trace_id(self.valid_trace_id), self.valid_span_id
+        )
+        new_span_context = self._extract(header)
+        self.assertEqual(new_span_context.trace_id, self.valid_trace_id)
+        self.assertEqual(new_span_context.span_id, self.valid_span_id)
+        self.assertEqual(new_span_context.trace_flags, TraceFlags(1))
+        self.assertTrue(new_span_context.is_remote)
+
+        header = "{}/{};o=10".format(
+            get_hexadecimal_trace_id(self.valid_trace_id), self.valid_span_id
+        )
+        new_span_context = self._extract(header)
+        self.assertEqual(new_span_context.trace_id, self.valid_trace_id)
+        self.assertEqual(new_span_context.span_id, self.valid_span_id)
+        self.assertEqual(new_span_context.trace_flags, TraceFlags(10))
+        self.assertTrue(new_span_context.is_remote)
+
+        header = "{}/{};o=0".format(
+            get_hexadecimal_trace_id(self.valid_trace_id), self.valid_span_id
+        )
+        new_span_context = self._extract(header)
+        self.assertEqual(new_span_context.trace_id, self.valid_trace_id)
+        self.assertEqual(new_span_context.span_id, self.valid_span_id)
+        self.assertEqual(new_span_context.trace_flags, TraceFlags(0))
+        self.assertTrue(new_span_context.is_remote)
+
+        header = "{}/{};o=0".format(
+            get_hexadecimal_trace_id(self.valid_trace_id), 345
+        )
+        new_span_context = self._extract(header)
+        self.assertEqual(new_span_context.trace_id, self.valid_trace_id)
+        self.assertEqual(new_span_context.span_id, 345)
+        self.assertEqual(new_span_context.trace_flags, TraceFlags(0))
+        self.assertTrue(new_span_context.is_remote)
```

Comment:
This set of assertions seem to test the same thing as the previous one.

Reply:
The difference is the length of span_id (20 vs 3)